File: packages/rook_sdk_health_connect/lib/src/data/mapper/background_read_status_mappers.dart
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk_health_connect/src/domain/enums/hc_background_read_status.dart';

extension BackgroundReadStatusMappers on BackgroundReadStatusProto {
  HCBackgroundReadStatus toDomain() {
    return switch (this) {
      BackgroundReadStatusProto.UNAVAILABLE =>
        HCBackgroundReadStatus.unavailable,
      BackgroundReadStatusProto.PERMISSION_NOT_GRANTED =>
        HCBackgroundReadStatus.permissionNotGranted,
      BackgroundReadStatusProto.PERMISSION_GRANTED =>
        HCBackgroundReadStatus.permissionGranted,
      _ => HCBackgroundReadStatus.permissionGranted,
    };
  }
}

File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/BackgroundReadStatusMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.enums.BackgroundReadStatus
import com.rookmotion.rook_sdk_health_connect.proto.BackgroundReadStatusProto

fun BackgroundReadStatus.toProto(): BackgroundReadStatusProto {
    return when (this) {
        BackgroundReadStatus.UNAVAILABLE -> BackgroundReadStatusProto.UNAVAILABLE
        BackgroundReadStatus.PERMISSION_NOT_GRANTED -> BackgroundReadStatusProto.PERMISSION_NOT_GRANTED
        BackgroundReadStatus.PERMISSION_GRANTED -> BackgroundReadStatusProto.PERMISSION_GRANTED
    }
}
